### PR TITLE
Np GL-1808 Allow RNAWithUMIs to take fastqs as input

### PR DESCRIPTION
**Pull request by nikellepetrillo**

### CODE REVIEW

File: pipelines/broad/rna_seq/RNAWithUMIsPipeline.wdl [hunk 1]
```diff
@@ -1,16 +1,27 @@
...
+		File? r2_fastq
 		String read1Structure
 		String read2Structure
 		File starIndex
 		String output_basename
 		File gtf
 
+		# only needed if inputs are fastqs instead of a bam
+		String? library_name
+        String? platform
```
Comment (kcibul) [inline pipelines/broad/rna_seq/RNAWithUMIsPipeline.wdl:20]: Spacing?

File: pipelines/broad/rna_seq/RNAWithUMIsPipeline.wdl [hunk 2]
```diff
@@ -19,9 +30,53 @@ workflow RNAWithUMIsPipeline {
...
+         input:
+           message = "Either bam or fastqs must be defined."
+      }
+    }
+
+    if (defined(r1_fastq) || defined(r2_fastq)) {
+      if ((defined(r1_fastq) && !defined(r2_fastq)) || (defined(r2_fastq) && !defined(r1_fastq))) {
+        call utils.ErrorWithMessage as ErrorMessageMissingR2Fastq {
+          input:
+            message = "R1_fastq and r2_fastq must both be defined"
```
Comment (gbggrant) [inline pipelines/broad/rna_seq/RNAWithUMIsPipeline.wdl:51]:
```suggestion
            message = "r1_fastq and r2_fastq must both be defined"
```

File: pipelines/broad/rna_seq/RNAWithUMIsPipeline.wdl [hunk 3]
```diff
@@ -19,9 +30,53 @@ workflow RNAWithUMIsPipeline {
...
+        }
+      }
+      if (!defined(library_name) || (!defined(platform)) || (!defined(platform_unit)) || (!defined(read_group_name)) || (!defined(sequencing_center))) {
+        call utils.ErrorWithMessage as ErrorMessageMissingFastqHeaderMetadata {
+          input:
+            message = "If r1_fastq and r2_fastq are defined then library_name, platform, platform_unit, read_group_name, and sequencing center must also be defined"
+        }
+      }
```
Comment (wjdingman1) [inline pipelines/broad/rna_seq/RNAWithUMIsPipeline.wdl:59]: May make sense to pull this out to its own task and just document the reasoning behind fastq vs ubam inputs

**Review by wjdingman1 — APPROVED** [review]
Looks good with minor comment!

File: pipelines/broad/rna_seq/RNAWithUMIsPipeline.wdl [hunk 4]
```diff
@@ -19,9 +30,53 @@ workflow RNAWithUMIsPipeline {
...
+
+    if (!defined(bam) && (!defined(r1_fastq) && !defined(r2_fastq))) {
+      call utils.ErrorWithMessage as ErrorMessageMissingInput {
+         input:
+           message = "Either bam or fastqs must be defined."
+      }
+    }
+
+    if (defined(r1_fastq) || defined(r2_fastq)) {
+      if ((defined(r1_fastq) && !defined(r2_fastq)) || (defined(r2_fastq) && !defined(r1_fastq))) {
```
Comment (gbggrant) [inline pipelines/broad/rna_seq/RNAWithUMIsPipeline.wdl:48]:
```suggestion
      if !(defined(r1_fast1q) && defined(r2_fastq)) {
```
I think that should work?  If not (both are defined)

File: pipelines/broad/rna_seq/RNAWithUMIsPipeline.wdl [hunk 5]
```diff
@@ -114,6 +169,50 @@ workflow RNAWithUMIsPipeline {
...
+        String library_name
+        String platform
+        String platform_unit
+        String read_group_name
+        String sequencing_center
+
+        Int disk = ceil(size(r2_fastq, "GiB") * 2.2)
+        Int cpu = 1
+        String docker = "us.gcr.io/broad-gotc-prod/picard-cloud:2.26.6"
+        Int memory_mb = 3850
```
Comment (gbggrant) [inline pipelines/broad/rna_seq/RNAWithUMIsPipeline.wdl:186]:
```suggestion
        Int memory_mb = 3750
```
Doesn't 3850 put us just over an N1? https://cloud.google.com/compute/all-pricing

File: pipelines/broad/rna_seq/RNAWithUMIsPipeline.wdl [hunk 6]
```diff
@@ -114,6 +169,50 @@ workflow RNAWithUMIsPipeline {
...
+        File r1_fastq
+        File r2_fastq
+        String output_basename
+        String library_name
+        String platform
+        String platform_unit
+        String read_group_name
+        String sequencing_center
+
+        Int disk = ceil(size(r2_fastq, "GiB") * 2.2)
```
Comment (gbggrant) [inline pipelines/broad/rna_seq/RNAWithUMIsPipeline.wdl:183]: maybe sum r1_fastq and r2_fastq here?  I would assume they are almost the same size, but can't be sure.

**Review by gbggrant — APPROVED** [review]
Some small changes suggested.

File: pipelines/broad/rna_seq/RNAWithUMIsPipeline.wdl (hunk 4 above)
Comment (nikellepetrillo) [reply pipelines/broad/rna_seq/RNAWithUMIsPipeline.wdl:48]: ah yes, a much more succinct way of putting it. this should work the same. although i think wdl will yell unless i put parens around that whole thing: ` if (!(defined(r1_fastq) && defined(r2_fastq))) {`
ill try it out

**Pull request merged**